### PR TITLE
M3-2353 Part II, State Management

### PR DESCRIPTION
**Pull request by rjstires**
## Description
Source Linode and details from Redux.

## Type of Change
- Non-breaking change

### CODE REVIEW

File: src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsLabelPanel.tsx [hunk 1]
```diff
@@ -126,10 +126,12 @@ const styled = withStyles(styles);
...
 
-const linodeContext = withLinode(context => ({
-  linodeId: context.data!.id,
-  linodeLabel: context.data!.label,
-  updateLinode: context.update
+const linodeContext = withLinode(({ linode }) => ({
+  linodeId: linode.id,
+  linodeLabel: linode.label,
+  updateLinode: () => {
+    window.alert(`deprecated updateLinode called`);
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsLabelPanel.tsx]: should be sending a Sentry error maybe?

File: src/features/linodes/LinodesDetail/LinodeSummary/SummaryPanel.tsx [hunk 2]
```diff
@@ -173,14 +173,16 @@ interface LinodeContextProps {
...
-  request: context.request
+const linodeContext = withLinode(({ linode }) => ({
+  linodeIpv4: linode.ipv4,
+  linodeIpv6: linode.ipv6,
+  linodeRegion: linode.region,
+  linodeImageId: linode.image,
+  linodeTags: linode.tags,
+  linodeId: linode.id,
+  request: () => {
+    window.alert(`depcreated request invoked`);
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodeSummary/SummaryPanel.tsx]: same as above?

File: src/features/linodes/LinodesDetail/LinodesDetail.container.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    }
+  };
+};
+
+const withLinode = connect(mapState);
+
+/** Gets the configs for the Linode on mount and on update when the linode ID has change. */
+const getConfigsForLinode = lifecycle<
+  OutterProps,
+  { configs: Linode.Config[] }
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodesDetail.container.ts]: You're never initializing configs anywhere, so this is technically `configs?: Linode.Config[]` - this means you'll also have to do null checking in the above HOC

File: src/features/linodes/LinodesDetail/LinodesDetail.container.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  const { entities: types, error: typesError } = state.__resources.types;
+  const { data: notifications = [] } = state.__resources.notifications;
+  const { events } = state.events;
+
+  const linode = linodes.find(l => l.id === linodeId);
+
+  const loading =
+    isLoading(state.__resources.linodes) || isLoading(state.__resources.types);
+
+  const error = linodesError || typesError;
```
Comment (johnwcallahan) [inline src/features/linodes/LinodesDetail/LinodesDetail.container.ts]: Any way we could make Linode Details still functional if `types` are unavailable? If you request-block `/types`, you'll see the giant, scary "Not found" error if you try to access Linode Details. This is in contrast to current production, which only bombs out if you try to access the "Resize" tab.

screen shot 2019-02-06 at 10 07 08 am

File: src/features/linodes/LinodesDetail/LinodesDetail.tsx [hunk 5]
```diff
@@ -464,7 +386,7 @@ class LinodeDetail extends React.Component<CombinedProps, State> {
         const errorStrings: string[] = errors.map(e => e.reason);
         /** "!" is okay because linode being undefined is being handled by render() */
```
Comment (johnwcallahan) [inline src/features/linodes/LinodesDetail/LinodesDetail.tsx]: Let's go ahead and remove this comment.

File: src/features/linodes/LinodesDetail/LinodesDetail.container.ts (hunk 4 above)
Comment (rjstires) [reply src/features/linodes/LinodesDetail/LinodesDetail.container.ts]: Yea I can swing that.

File: src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsLabelPanel.tsx (hunk 1 above)
Comment (rjstires) [reply src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsLabelPanel.tsx]: These are being resolved in Part III where I pass actions.

File: src/features/linodes/LinodesDetail/LinodeSummary/SummaryPanel.tsx (hunk 2 above)
Comment (rjstires) [reply src/features/linodes/LinodesDetail/LinodeSummary/SummaryPanel.tsx]: > These are being resolved in Part III where I pass actions.

File: src/features/linodes/LinodesDetail/LinodesDetail.container.ts (hunk 3 above)
Comment (rjstires) [reply src/features/linodes/LinodesDetail/LinodesDetail.container.ts]: Great catch! Resolved in next PR.

**Pull request merged**